tests/integration_tests/functional/test_drive_vhost_user.py

```diff
@@ -30,6 +37,52 @@ def spawn_vhost_user_backend(vm, host_mem_path, socket_path):
     return proc
 
 
+@pytest.fixture
+def rootfs_ubuntu_22_partuuid_and_disk_path(rootfs_ubuntu_22):
+    """
+    We create new file on the host, get it's partuuid and use it as a rootfs.
+    It is important to create it on the host, because qemu vhost-user-block
+    can't serve files from tmpfs.
+    """
+    disk_img = LOCAL_BUILD_PATH / "img" / "tmp_disk.img"
+    initial_size = rootfs_ubuntu_22.stat().st_size + 50 * MB
+    disk_img.touch()
+    os.truncate(disk_img, initial_size)
+    check_output(f"echo type=83 | sfdisk {str(disk_img)}", shell=True)
+    stdout = check_output(
+        f"losetup --find --partscan --show {str(disk_img)}", shell=True
+    )
+    loop_dev = stdout.decode("ascii").strip()
+    check_output(f"dd if={str(rootfs_ubuntu_22)} of={loop_dev}p1", shell=True)
+
+    # UUID=$(sudo blkid -s UUID -o value "${loop_dev}p1")
+    stdout = check_output(f"blkid -s PARTUUID -o value {loop_dev}p1", shell=True)
+    partuuid = stdout.decode("ascii").strip()
+
+    # cleanup: release loop device
+    check_output(f"losetup -d {loop_dev}", shell=True)
+
+    yield (partuuid, disk_img)
+    disk_img.unlink()
+
+
+def _check_block_size(ssh_connection, dev_path, size):
```

Comment:
can we extract these helpers into utils?

Reply:
I don't think it is worth it. The function is 3 lines long.